bundles/ch.unibe.iam.scg.archie/src/ch/unibe/iam/scg/archie/controller/TableFactory.java

```diff
@@ -93,6 +102,88 @@ public TableViewer createTableFromData(final Composite parent, final DataSet dat
 		return tableViewer;
 	}
 
+	/**
+	 * Creates a TreeViewer from the given DataSet. The method supports two
+	 * modes: a grouped view by (service), where data is displayed
+	 * hierarchically with expandable parent nodes, or a flat view without
+	 * grouping.
+	 *
+	 * The table columns are generated dynamically based on the DataSet
+	 * headings. Additionally, column sorting is enabled for each column.
+	 *
+	 * @param parent
+	 *            Composite that holds the TreeViewer.
+	 * @param dataset
+	 *            DataSet providing the headings and content.
+	 * @param labelProvider
+	 *            Label provider used to render each cell.
+	 * @param contentProvider
+	 *            Content provider supplying the data structure.
+	 * @param groupBy
+	 *            If true, the data will be grouped by Leistung (hierarchical
+	 *            view) and an arrow column is added for expand/collapse. If
+	 *            false, the data is shown flat without grouping.
+	 * @return Configured TreeViewer instance.
+	 */
+	public TreeViewer createTreeFromData(final Composite parent, final DataSet dataset,
+			final ILabelProvider labelProvider, final IContentProvider contentProvider, boolean groupBy) {
+
+		// Create base Tree widget with scrolling and selection behavior
+		Tree tree = new Tree(parent, SWT.H_SCROLL | SWT.V_SCROLL | SWT.FULL_SELECTION | SWT.MULTI);
+		tree.setHeaderVisible(true);
+		tree.setLinesVisible(true);
+
+		// Set layout manager for column resizing
+		TreeColumnLayout treeLayout = new TreeColumnLayout();
+		parent.setLayout(treeLayout);
+
+		// Create columns dynamically based on headings
+		createTreeColumns(tree, treeLayout, dataset.getHeadings(), groupBy);
+
+		// Create TreeViewer and set content & label providers
+		TreeViewer treeViewer = new TreeViewer(tree);
+		LeistungenTreeContentProvider contentProv = new LeistungenTreeContentProvider(groupBy);
```

Comment:
In einer allgemein gehaltenen `TableFactory` eine `LeistungenTreeContentProvider` zu verwenden geht nicht.